src/main/java/com/commercetools/sync/categories/CategorySync.java

```diff
@@ -186,11 +187,13 @@ private int getNumberOfProcessedCategories(@Nonnull final List<CategoryDraft> ca
         final int numberOfNullCategoryDrafts = categoryDrafts.stream()
                                                              .filter(Objects::isNull)
                                                              .collect(Collectors.toList()).size();
-        final int numberOfCategoryDraftsNotProcessedBefore = categoryDrafts.stream()
-                                       .filter(Objects::nonNull)
-                                       .map(CategoryDraft::getKey)
-                                       .filter(categoryDraftKey -> !processedCategoryKeys.contains(categoryDraftKey))
-                                       .collect(Collectors.toList()).size();
+        final int numberOfCategoryDraftsNotProcessedBefore =
+            categoryDrafts.stream()
+                          .filter(Objects::nonNull)
+                          .map(CategoryDraft::getKey)
+                          .filter(categoryDraftKey ->
+                              categoryDraftKey == null || !processedCategoryKeys.contains(categoryDraftKey))
```

Comment:
This null check was made to avoid hitting an NPE in `processedCategoryKeys.contains(categoryDraftKey)` if `categoryDraftKey` is `null`. Since `processedCategoryKeys` is now a `KeySetView` and this throws an NPE on it's contains method if the passed arg is null, as opposed to what it was previously (`HashSet`).